lib/database/migrations/20240201102034-create-quality-control-flags-tables.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+    up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(async (transaction) => {
+        await queryInterface.createTable('quality_control_flag_reasons', {
+            // Properties
+            id: {
+                type: Sequelize.INTEGER,
+                allowNull: false,
+                primaryKey: true,
+                autoIncrement: true,
+            },
+            name: {
+                type: Sequelize.STRING,
+                allowNull: false,
+                unique: true,
+            },
+            method: {
+                type: Sequelize.STRING,
+                unique: true,
+                allowNull: false,
+            },
+            bad: {
+                type: Sequelize.BOOLEAN,
+                allowNull: false,
+            },
+            archived_at: {
+                type: Sequelize.DATE,
+                allowNull: true,
+            },
+
+            // Timestamps
+            created_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP'),
+            },
+            updated_at: {
+                allowNull: false,
+                type: Sequelize.DATE,
+                defaultValue: Sequelize.literal('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'),
+            },
+        }, { transaction });
+
+        await queryInterface.createTable('quality_control_flags', {
+            // Properties
+            id: {
+                type: Sequelize.INTEGER,
+                allowNull: false,
+                primaryKey: true,
+                autoIncrement: true,
+            },
+            from: {
+                type: Sequelize.DATE,
+                allowNull: false,
+            },
+            to: {
+                type: Sequelize.DATE,
+                allowNull: false,
+            },
+            comment: {
+                type: Sequelize.TEXT,
+            },
+            provenance: {
+                type: Sequelize.ENUM('HUMAN', 'SYNC', 'ASYNC', 'MC'),
+                allowNull: false,
+            },
+
+            // Associations
+            user_id: {
```

Comment:
I think it would make more sense to name it `created_by_id`, so that we know the relation.

Reply:
applied